### PR TITLE
CIR-591  Updated Schema for UkCompanies Restrictions and Reactivations + Misc

### PR DESCRIPTION
**Pull request by mywyau**

### CODE REVIEW

File: public/api/conf/1.0/schemas/dereferenced/appointReportingCompanySchema.json [hunk 1]
```diff
@@ -39,13 +39,6 @@
                     "maxLength": 10,
```
Comment (LadonJackson) [inline public/api/conf/1.0/schemas/dereferenced/appointReportingCompanySchema.json]: There's another "nonUkCrn" in this file which I think can be removed

File: public/api/conf/1.0/schemas/dereferenced/revokeReportingCompanySchema.json [hunk 2]
```diff
@@ -39,13 +39,6 @@
                     "maxLength": 10,
```
Comment (LadonJackson) [inline public/api/conf/1.0/schemas/dereferenced/revokeReportingCompanySchema.json]: Another one in here too

File: public/api/conf/1.0/schemas/subSchemas/accountingPeriod.json [hunk 3]
```diff
@@ -4,12 +4,12 @@
     "startDate": {
       "description": "The start date of the accounting period",
       "type": "string",
-      "format": "date"
```
Comment (MJCallahanPage) [inline public/api/conf/1.0/schemas/subSchemas/accountingPeriod.json]: I'd leave this as format date. Even though our validation library doesn't support it, it's better JSON Schema documentation

File: public/api/conf/1.0/schemas/subSchemas/accountingPeriod.json [hunk 4]
```diff
@@ -4,12 +4,12 @@
...
+      "pattern": "^((((19|[2-9]\\d)\\d{2})\\-(0[13578]|1[02])\\-(0[1-9]|[12]\\d|3[01]))|(((19|[2-9]\\d)\\d{2})\\-(0[13456789]|1[012])\\-(0[1-9]|[12]\\d|30))|(((19|[2-9]\\d)\\d{2})\\-02\\-(0[1-9]|1\\d|2[0-8]))|(((1[6-9]|[2-9]\\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))\\-02\\-29))$"
     },
     "endDate": {
       "description": "The end date of the accounting period",
       "type": "string",
-      "format": "date"
```
Comment (MJCallahanPage) [inline public/api/conf/1.0/schemas/subSchemas/accountingPeriod.json]: Same with this

File: public/api/conf/1.0/schemas/subSchemas/ukCompanyFull.json [hunk 5]
```diff
@@ -25,23 +25,40 @@
...
+            "description": "The Net Tax Interest Income amount",
+            "type": "number",
+            "maximum": 0
+          }
+        },
+        "required": [
+          "netTaxInterestIncome"
+        ]
+      }
+    },
```
Comment (MJCallahanPage) [inline public/api/conf/1.0/schemas/subSchemas/ukCompanyFull.json]: How does this work, what's the thinking for it?

**Pull request merged**